src/radical/entk/task.py

```diff
@@ -29,1135 +114,229 @@ class Task(object):
     #
-    @property
-    def uid(self):
-        '''
-        Unique ID of the current task
+    def _post_verifier(self, k, v):
+
+        if not v:
+            return
+
+        if k in ['uid', 'name']:
+            invalid_symbols = punctuation.replace('.', '')
+            if any(symbol in v for symbol in invalid_symbols):
+                raise ree.EnTKError(
+                    'Incorrect symbol for attribute "%s" (%s). %s' %
+                    (k, v, NAME_MESSAGE))
+
+        elif k == 'state':
+            if v not in res._task_state_values:
+                raise ree.ValueError(
+                    obj=self['uid'],
+                    attribute=k,
+                    expected_value=list(res._task_state_values),
+                    actual_value=v)
+            self['state_history'].append(v)
+
+        elif k == 'state_history':
+            for _v in v:
+                if _v not in res._task_state_values:
+                    raise ree.ValueError(
+                        obj=self['uid'],
+                        attribute='state_history element',
+                        expected_value=list(res._task_state_values),
+                        actual_value=_v)
+
+        elif k == 'tags':
+            if list(v) != ['colocate']:
+                raise ree.EnTKError(
+                    'Incorrect structure for attribute "%s" of object %s' %
+                    (k, self['uid']))
 
-        :getter: Returns the unique id of the current task
-        :type: String
-        '''
-        return self._uid
+    # --------------------------------------------------------------------------
+    #
+    def _verify_setter(self, k, v):
+        v = super()._verify_setter(k, v)
+        self._post_verifier(k, v)
+        return v
 
+    # --------------------------------------------------------------------------
+    #
+    def _verify(self):
+        for k, v in self.items():
+            self._post_verifier(k, v)
 
+    # --------------------------------------------------------------------------
+    #
     @property
     def luid(self):
-        '''
+        """
         Unique ID of the current task (fully qualified).
-
         example:
-            >>> task.luid
+            > task.luid
             pipe.0001.stage.0004.task.0234
-
-        :getter: Returns the fully qualified uid of the current task
+        :luid: Returns the fully qualified uid of the current task
         :type: String
-        '''
+        """
 
         # TODO: cache
 
+        p_elem = self['parent_pipeline'].get('name')
+        s_elem = self['parent_stage'].get('name')
+        t_elem = self['name']
 
-        p_elem = self.parent_pipeline.get('name')
-        s_elem = self.parent_stage.get('name')
-        t_elem = self.name
-
-        if not p_elem: p_elem = self.parent_pipeline['uid']
-        if not s_elem: s_elem = self.parent_stage['uid']
-        if not t_elem: t_elem = self.uid
+        if not p_elem: p_elem = self['parent_pipeline']['uid']
```

Comment:
Why did that change actually - shouldn't `Munch` keep this valid code?

Reply:
since it is internal calls, I wanted to decrease number of attribute-resolving calls: accessing items is faster than accessing attributes